zwave_js_server/model/node/__init__.py

```diff
@@ -348,54 +349,57 @@ def keep_awake(self) -> bool:
 
     def update(self, data: NodeDataType) -> None:
         """Update the internal state data."""
-        self.data = data
+        self.data = copy.deepcopy(data)
         self._device_config = DeviceConfig(self.data.get("deviceConfig", {}))
         self._statistics = NodeStatistics(
             self.client, self.data.get("statistics", DEFAULT_NODE_STATISTICS)
         )
 
-        # Remove stale values
-        value_ids = (_get_value_id_str_from_dict(self, val) for val in data["values"])
-        self.values = {
-            value_id: val
-            for value_id, val in self.values.items()
-            if value_id in value_ids
+        new_values_data = {
+            _get_value_id_str_from_dict(self, val): val
+            for val in self.data.pop("values")
         }
+        new_value_ids = set(new_values_data)
+
+        # Remove stale values
+        for value_id in (stale_value_ids := set(self.values) - new_value_ids):
```

Comment:
I don't see the value in a walrus here. We'll always use the local variable.

Reply:
Walrus is because it's referenced in the next code block to get the remaining ids to update/create. It may be more clear that's the case with the last commit rather than setting it using a walrus

Reply:
Right, but we don't need to use a walrus. Use a separate line above.